code/datums/diseases/asthma_attack.dm

```diff
@@ -0,0 +1,262 @@
+	severity = DISEASE_SEVERITY_HARMFUL
+	stage_prob = 5
+
+	max_time_til_remission = 80 SECONDS
+	min_time_til_remission = 120 SECONDS
+	max_stages = 4
+
+	cure_chance = 25
+
+	stage_to_inflammation_per_second = list(
+		"2" = 1,
+		"3" = 2,
+		"4" = 4,
+	)
+
+/datum/disease/asthma_attack/moderate/stage_act(seconds_per_tick, times_fired)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	if (SPT_PROB(15, seconds_per_tick))
+		to_chat(affected_mob, span_warning(pick("Mucous runs down the back of your throat.", "You swallow excess mucus.")))
+
+	if (stage >= 4)
+		if (SPT_PROB(10, seconds_per_tick))
+			to_chat(affected_mob, span_warning("You briefly choke on the mucus piling in your throat!"))
+			affected_mob.losebreath++
+
+
+/datum/disease/asthma_attack/severe
+	severity = DISEASE_SEVERITY_DANGEROUS
+	stage_prob = 6
+
+	max_time_til_remission = 120 SECONDS
+	min_time_til_remission = 80 SECONDS
+	max_stages = 5
+
+	cure_chance = 25
+
+	stage_to_inflammation_per_second = list(
+		"2" = 1,
+		"3" = 3,
+		"4" = 6,
+		"5" = 8,
+	)
+
+	visibility_flags = HIDDEN_SCANNER
+	alert_ghosts = TRUE
+
+/datum/disease/asthma_attack/severe/stage_act(seconds_per_tick, times_fired)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	if (stage > 1)
+		visibility_flags &= ~HIDDEN_SCANNER // revealed
+
+	if (SPT_PROB(15, seconds_per_tick))
+		to_chat(affected_mob, span_warning(pick("Mucous runs down the back of your throat.", "You swallow excess mucus.")))
+	else if (SPT_PROB(20, seconds_per_tick))
+		affected_mob.emote("cough")
+
+	if (stage >= 4)
+		if (SPT_PROB(15, seconds_per_tick))
+			to_chat(affected_mob, span_warning("You briefly choke on the mucus piling in your throat!"))
+			affected_mob.losebreath++
+
+/datum/disease/asthma_attack/critical
+	severity = DISEASE_SEVERITY_BIOHAZARD
+	stage_prob = 85
+
+	max_time_til_remission = 60 SECONDS // this kills you extremely quickly, so its fair
+	min_time_til_remission = 40 SECONDS
+	max_stages = 6
+
+	cure_chance = 30
+
+	stage_to_inflammation_per_second = list(
+		"1" = 5,
+		"2" = 6,
+		"3" = 7,
+		"4" = 10,
+		"5" = 20,
+		"6" = 500, // youre fucked frankly
+	)
+
+	/// Have we warned our user of the fact they are at stage 5? If no, and are at or above stage five, we send a warning and set this to true.
+	var/warned_user = FALSE
+	/// Have we ever reached our max stage? If no, and we are at our max stage, we send a ominous message warning them of their imminent demise.
+	var/max_stage_reached = FALSE
+
+/datum/disease/asthma_attack/critical/stage_act(seconds_per_tick, times_fired)
+	. = ..()
+	if (!.)
+		return FALSE
+
+	if (stage < 5)
+		if (SPT_PROB(75, seconds_per_tick))
+			to_chat(affected_mob, span_warning(pick("Mucous runs down the back of your throat.", "You swallow excess mucus.")))
+
+	var/wheeze_chance
+	if (!warned_user && stage >= 5)
+		to_chat(affected_mob, span_userdanger("You feel like your lungs are filling with fluid! It's getting incredibly hard to breathe!"))
+		warned_user = TRUE
+
+	switch (stage)
+		if (1)
```

Comment:
These should be defines

Reply:
why? its pretty self explanatory what these are, theyre jsut the stages